## Data : BreastCancer

### Response variable : Class  and Independet variable are Clump, UniCell_Size ,..... Mitloses.
    

In [2]:

import pandas as pd
import numpy as np
import warnings;warnings.simplefilter('ignore')

brcan=pd.read_csv("E:/class/R _Cource/dataset class/BreastCancer.csv")
brcan.head()


Code  Clump  UniCell_Size  Uni_CellShape  MargAdh  SEpith  BareN  \
0   61634      5             4              3        1       2      2   
1   63375      9             1              2        6       4     10   
2   76389     10             4              7        2       2      8   
3   95719      6            10             10       10       8     10   
4  128059      1             1              1        1       2      5   

   BChromatin  NoemN  Mitoses      Class  
0           2      3        1     Benign  
1           7      7        2  Malignant  
2           6      1        1  Malignant  
3           7     10        7  Malignant  
4           5      1        1     Benign

In [7]:
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score

X = brcan.iloc[:,1:10]
y = brcan.iloc[:,10]

from sklearn.model_selection import train_test_split 

# Create training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, 
                                                    test_size = 0.3, 
                                                    random_state=42,
                                                    stratify=y)

kfold = KFold(n_splits=5, random_state=42)


In [8]:

#kfold = StratifiedKFold(n_splits=5, random_state=42)
from sklearn.linear_model import LogisticRegression
logreg = LogisticRegression()


# Fit the classifier to the training data
logreg.fit(X_train,y_train)

logreg.fit(X,y)
logreg.coef_
logreg.intercept_
print(np.exp(logreg.coef_))



[[1.3094436  1.17790289 1.30893008 1.13980438 0.94270509 1.43662367
  1.20680096 1.1576484  1.25455165]]


In [9]:

from sklearn.model_selection import KFold
from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import  accuracy_score
from sklearn.metrics import  roc_auc_score



results = cross_val_score(logreg, X, y, cv=kfold, scoring='roc_auc')
print(results)
print("ROC AUC: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))



[0.9952627  0.97844008 0.99908488 0.99609629 0.9969419 ]
ROC AUC: 99.32% (0.75%)


### Accuracy score is 99.32 % for k is equal to 5

In [10]:


###  accuracy

results = cross_val_score(logreg, X, y, cv=kfold, scoring='accuracy')
print(results)
print("accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

### AUC AND accuracy is nearly 96 %



[0.96428571 0.90714286 0.97142857 0.97857143 0.99280576]
accuracy: 96.28% (2.94%)


### AUC  value is 96.28 % both AUC and Accuracy are nearly 96%